moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -433,7 +433,12 @@ void MotionPlanningFrame::changePlanningGroup()
 void MotionPlanningFrame::sceneUpdate(planning_scene_monitor::PlanningSceneMonitor::SceneUpdateType update_type)
 {
   if (update_type & planning_scene_monitor::PlanningSceneMonitor::UPDATE_GEOMETRY)
+  {
     planning_display_->addMainLoopJob(boost::bind(&MotionPlanningFrame::populateCollisionObjectsList, this));
+    // Enable Clear Octomap button only if octomap exists in the world
+    const planning_scene_monitor::LockedPlanningSceneRO& ps = planning_display_->getPlanningSceneRO();
+    ui_->clear_octomap_button->setEnabled(ps->getWorld()->hasObject(ps->OCTOMAP_NS));
+  }
```

Comment:
Is this called very frequently? It might not be worth the cost of updating the button, if so

Reply:
Only when the scene geometry changes. If this impacted performance I would think that `populateCollisionObjectsList` has a larger effect than updating the button.

Reply:
(working on testing this now)

Sorry, what does this mean exactly:  `Only when the scene geometry changes`. Is that every time a new point cloud arrives (if octomap is enabled)? Or just when a new object is added to the scene?

Reply:
That's supposed to mean whenever an object is added/removed/changed in the PlanningScene's `world` (= observers are called), but since afaik the octomap is also an object in the world (`OCTOMAP_NS`), I guess it's yes to both.